### PR TITLE
skip validation and defaulting for CAPI release

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards: https://github.com/giantswarm/giantswarm/issues/16119
This just skips Clusters and Machinedeployments that are in a CAPI release version. We might want to refine it more in the future.

### CODE REVIEW

**Comment by MarcelMue** [comment]
Should we merge this or keep it in a branch for now?

**Comment by anvddriesch** [comment]
I don't see a reason why we can not merge it. But I'm also fine with just deploying this branch to test installations.

File: pkg/aws/common.go
Comment (njuettner) [inline pkg/aws/common.go:26]: I'd suggest to rename it CAPIRelease to make it more obvious

File: pkg/aws/common.go
Comment (njuettner) [inline pkg/aws/common.go:80]: we should handle all errors right?

File: pkg/aws/cluster/mutate_cluster.go
Comment (njuettner) [inline pkg/aws/cluster/mutate_cluster.go:119]: could we circumvent the error handling by just checking if it's capi or not to make it simple?
We could also check if if the microerror is kind "..." and than return
}

Comment (anvddriesch) [reply pkg/aws/cluster/mutate_cluster.go:119]: You mean we could just ignore the error here?

**Comment by anvddriesch** [comment]
I made the changes suggested in the review.
Still, for now I think we can just leave this as a branch that is deployed on test installations until we figure out how we want to manage running CAPI and GS releases together.

File: pkg/aws/cluster/mutate_cluster.go
Comment (njuettner) [reply pkg/aws/cluster/mutate_cluster.go]: Hey 🙂 , I was thinking either we could return only the error when calling `aws.IsCAPIRelease` and check the err or just only return the bool.

Comment (anvddriesch) [reply pkg/aws/cluster/mutate_cluster.go]: Ahhh okay got it :D Sorry for auf dem Schlauch stehen.

**Comment by njuettner** [comment]
There's also an configuration for admission controller to select webhook configs based on labels, in case this is interesting https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-objectselector

**Comment by anvddriesch** [comment]
This would be very interesting to make a webhook active only for certain labels. So we could use it if we add some labels in the future to show something is CAPI or not CAPI.
However, I don't think this could work right now for label values smaller than something, right? (e.g only use webhook for releases<v20.0.0

**Comment by njuettner** [comment]
> This would be very interesting to make a webhook active only for certain labels. So we could use it if we add some labels in the future to show something is CAPI or not CAPI.
> However, I don't think this could work right now for label values smaller than something, right? (e.g only use webhook for releases<v20.0.0

You're right, we still need your implementation, just only added it as something we should take a look later at some point 🙂 .

File: pkg/aws/common.go
Comment (kubasobon) [inline pkg/aws/common.go:26]:
```suggestion
	// FirstCAPIRelease is the last GS release that does not run on Giant Swarm controllers
```

**Review by MarcelMue — APPROVED** [review]

**Comment by JosephSalisbury** [comment]
@anvddriesch what stops us merging this?

**Comment by anvddriesch** [comment]
Nothing. I'll merge it.

**Pull request merged**
